specs/Weather/WeatherObserved/harvesters/portugal/portugal_weather_observed.py

```diff
@@ -0,0 +1,567 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+    This program collects Portugal weather observation from IPMA and uploads it to the Orion Context Broker.
+    It upload the list of stations on-fly from
+      - http://api.ipma.pt/open-data/observation/meteorology/stations/obs-surface.geojson.
+
+    Legal notes:
+      - http://www.ipma.pt/en/siteinfo/index.html?page=index.xml
+
+    Examples:
+      - get the weather observation from IPMA:
+        curl -X GET --header 'Accept: application/json' \
+            'https://api.ipma.pt/open-data/observation/meteorology/stations/observations.json'
+
+    AsyncIO name convention:
+    async def name - entry point for asynchronous data processing/http requests and post processing
+    async def name_bounded - intermediate step to limit amount of parallel workers
+    async def name_one - worker process
+"""
+
+from aiohttp import ClientSession, client_exceptions
+from argparse import ArgumentTypeError, ArgumentParser
+from asyncio import Semaphore, ensure_future, gather, run
+from copy import deepcopy
+from datetime import datetime
+from pytz import timezone
+from re import sub
+from sys import stdout
+from time import sleep
+from yajl import dumps, loads
+from yaml import safe_load as load
+from requests import get, exceptions
+import logging
+
+default_latest = False                # preserve only latest values
+default_limit_entities = 50           # amount of entities per 1 request to Orion
+default_limit_target = 50             # amount of parallel request to Orion
+default_log_level = 'INFO'
+default_orion = 'http://orion:1026'   # Orion Contest Broker endpoint
+default_path = '/Portugal'            # header FIWARE-SERVICEPATH
+default_service = 'weather'           # header FIWARE-SERVICE
+default_timeout = -1                  # if value != -1, then work as a service
+
+http_ok = [200, 201, 204]
```

Comment:
can we separate here constants by different concerns? i.e. just a blank line between groups of them

Reply:
same apply for the rest of harvesters

Reply:
Maybe it's better to avoid a blank line? Just an alphabet order?